src/arch/ArchHooks/ArchHooks_Win32Static.cpp

```diff
@@ -8,44 +10,37 @@
 
 #include <cstdint>
 #include <vector>
-
-// for QueryPerformanceCounter
 #include <windows.h>
 #include <mmsystem.h>
+
 #if defined(_MSC_VER)
 #pragma comment(lib, "winmm.lib")
 #endif
 
-static bool g_bTimerInitialized;
+constexpr std::int64_t MICROSECONDS_IN_SECOND = 1000000;
+
+static bool was_timer_initialized = false;
 
 /* QueryPerformanceCounter variables below.
  * QueryPerformanceCounter and QueryPerformanceFrequency expect
  * a LARGE_INTEGER, which is a union. These functions store data
  * in the QuadPart of the LARGE_INTEGER, which is a 64-bit integer. */
-LARGE_INTEGER g_liFrequency, g_liStartTime, g_liCurrentTime;
+static LARGE_INTEGER ticks_per_second;
```

Comment:
I wanted to make it clear that the value of this isn't supposed to change, although I do recognize it's safe enough with the static boolean.

Reply:
... static doesn't imply that the values don't change?

Put these variables in an anonymous namespace and then drop the static qualifier.

Reply:
I felt static is more meaningful to represent the lifetime of the variable when there is a static boolean to prevent accidental re-initialization,  but not as appropriate with the current method.